### PR TITLE
Feature 553/refactor role exceptions closes #553

### LINKED ISSUE #553: BE> Refactor Role to use common exceptions (S)
Definition of Done

 - [x] refactor controller to use common handlers and exceptions
 - [x] refactor impl to use common exceptions
 - [x] remove unused specific exception classes
 - [x] update tests as needed

### PR DESCRIPTION
**Pull request by hwilliams314**

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/services/role/RoleServicesImpl.java [hunk 1]
```diff
@@ -20,46 +21,44 @@ public RoleServicesImpl(RoleRepository roleRepo,
...
+            throw new BadArgException(String.format("Invalid role %s", role));
+        } else if (role.getId() != null) {
+            throw new BadArgException(String.format("Found unexpected id %s for role", role.getId()));
+        } else if (memberRepo.findById(memberId).isEmpty()) {
+            throw new BadArgException(String.format("Member %s doesn't exist", memberId));
+        } else if (roleRepo.findByRoleAndMemberid(roleType, role.getMemberid()).isPresent()) {
+            throw new BadArgException(String.format("Member %s already has role %s", memberId, roleType));
         }
+
+        roleRet = roleRepo.save(role);
```
Comment (Bagurp) [inline server/src/main/java/com/objectcomputing/checkins/services/role/RoleServicesImpl.java:39]: We could remove the variable `roleRet` and return this directly

File: server/src/main/java/com/objectcomputing/checkins/services/role/RoleServicesImpl.java [hunk 2]
```diff
@@ -20,46 +21,44 @@ public RoleServicesImpl(RoleRepository roleRepo,
...
+
+        if (roleType == null || memberId == null) {
+            throw new BadArgException(String.format("Invalid role %s", role));
+        } else if (id == null || roleRepo.findById(id).isEmpty()) {
+            throw new BadArgException(String.format("Unable to locate role to update with id %s", id));
+        } else if (memberRepo.findById(memberId).isEmpty()) {
+            throw new BadArgException(String.format("Member %s doesn't exist", memberId));
         }
+
+        roleRet = roleRepo.update(role);
```
Comment (Bagurp) [inline server/src/main/java/com/objectcomputing/checkins/services/role/RoleServicesImpl.java:61]: same here

**Review by Bagurp — APPROVED** [review]

File: server/src/main/java/com/objectcomputing/checkins/services/role/RoleServicesImpl.java (hunk 1 above)
Comment (hwilliams314) [reply server/src/main/java/com/objectcomputing/checkins/services/role/RoleServicesImpl.java]: done

File: server/src/main/java/com/objectcomputing/checkins/services/role/RoleServicesImpl.java (hunk 2 above)
Comment (hwilliams314) [reply server/src/main/java/com/objectcomputing/checkins/services/role/RoleServicesImpl.java]: done

**Review by mkimberlin — APPROVED** [review]

**Pull request merged**